iamkanhaiyakumar/ai-content-generator
# Feat loadingUI

## What does this PR do?

This PR introduces a loading animation component (`LoadingUI`) that provides users with a visual indicator of progress while the main content is being loaded. The loading UI displays a bouncing image and a progress bar with a percentage indicator. This update improves user experience by making it clear when content is loading.

Fixes #80 

### Test Video

https://github.com/user-attachments/assets/910a47a9-6cf3-4780-9f77-3774b81b8d94

Desktop View:
![image](https://github.com/user-attachments/assets/0b6d5b91-a2a0-4528-94d9-0f3217801a49)

Mobile View:
![image](https://github.com/user-attachments/assets/de4151fa-add5-4210-b01b-9941ce26f18f)


## Type of change


## How should this be tested?

To test this PR:
1. Start the application and observe the loading UI during initial render.
2. Confirm that the progress bar increments smoothly to 100%.
3. Verify that, upon reaching 100%, the loading animation is hidden and the main content is displayed.
4. Test responsiveness and ensure the loading UI adapts to various screen sizes.

### Test Cases:
- [ ] **Test A:** Ensure the loading animation and progress bar are displayed when the component mounts.
- [ ] **Test B:** Confirm that the progress bar fills incrementally to 100%, hiding the loading animation afterward.
- [ ] Incase of any issue, try reloading

## Mandatory Tasks

- [x] Self-reviewed the code for consistency, responsiveness, and correct implementation of the loading animation.
- [x] Ensured smooth transition between loading UI and main content.
